src/prepared_geometry.rs

```diff
@@ -37,18 +41,28 @@ impl PreparedGeometry {
     ///                      .expect("Invalid geometry");
     /// let prepared_geom = PreparedGeometry::new(&geom1);
     /// ```
-    pub fn new<G: Geom>(g: &G) -> GResult<PreparedGeometry> {
+    pub fn new<G: Geom>(g: &'a G) -> GResult<PreparedGeometry<'a>> {
         with_context(|ctx| unsafe {
             let ptr = GEOSPrepare_r(ctx.as_raw(), g.as_raw());
             PreparedGeometry::new_from_raw(ptr, ctx, "new")
         })
     }
 
+    pub fn new_owning(g: Geometry) -> GResult<PreparedGeometry<'static>> {
```

Comment:
Not a big fan of this but I suppose it's for the case where someone wants to return both a `Geometry` and a `PreparedGeometry` right? If so I don't think it's a case that we should allow. There are many ways to handle it and I don't think returning both types from a function is a good Rust use case.

Reply:
Yes, it's for case when someone wants to be able to move `PreparedGeometry` freely around, encapsulate in inside some struct etc. I think it's valid case, not something that should be difficult or impossible to do. I have posted some examples in the issue. I don't see the other ways to handle it (or perhaps the answer is to not encapsulate `PreparedGeometry` anywhere, but that sound very limiting.)